ShinerNFT.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import "erc721a/contracts/ERC721A.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
+contract OwnableDelegateProxy {}
+contract ProxyRegistry { mapping(address => OwnableDelegateProxy) public proxies; }
+
+contract LocalTest is ERC721A, Ownable
+{
+	uint256 public constant MAX_SUPPLY = 11111;
+	uint256 public constant MINT_PRICE = 0.069 ether;
+	address internal constant _community = 0x36E92eF696ac7f7B7Cf7BA9EC0C7D42E6e65A8AD;
+	string internal _baseTokenURI = 'https://fake.robocosmic.online/';
+	address immutable proxyRegistryAddress;
+	bytes32 public _merkleRoot;
+
+	enum Status { Initial, Presale, Sale, SoldOut }
+	Status public status = Status.Initial;
+
+	event StatusChanged();
+
+	constructor(
+		bytes32 merkleRoot,
+		address _proxyRegistryAddress
+	) ERC721A("LocalTest", "LTS")
+	{
+		_merkleRoot = merkleRoot;
+		proxyRegistryAddress = _proxyRegistryAddress;
+
+		_mint(_community, 13, '', false); // Initial Mint
+	}
+
+	modifier onlyAccounts() { require(msg.sender == tx.origin, "NO_ALLOWED_ORIGIN"); _; }
+
+	function mint(uint256 quantity)
+		external
+		payable
+		onlyAccounts
+	{
+		require(status == Status.Sale, "SALE_CLOSED");
+		require(quantity != 0, "INVALID_QUANTITY");
```

Comment:
Not necessary, as `_mint` will revert if `quantity` is 0.

You can add a check on client side if you want the UI to be user friendly.